File: MPChartLib/src/main/java/com/github/mikephil/charting/formatter/DefaultAxisValueFormatter.java
package com.github.mikephil.charting.formatter;

import com.github.mikephil.charting.components.AxisBase;

import java.text.DecimalFormat;

/**
 * Created by philipp on 02/06/16.
 */
public class DefaultAxisValueFormatter implements AxisValueFormatter {

    /**
     * decimalformat for formatting
     */
    protected DecimalFormat mFormat;

    /**
     * the number of decimal digits this formatter uses
     */
    protected int digits = 0;

    /**
     * Constructor that specifies to how many digits the value should be
     * formatted.
     *
     * @param digits
     */
    public DefaultAxisValueFormatter(int digits) {
        this.digits = digits;

        StringBuffer b = new StringBuffer();
        for (int i = 0; i < digits; i++) {
            if (i == 0)
                b.append(".");
            b.append("0");
        }

        mFormat = new DecimalFormat("###,###,###,##0" + b.toString());
    }

    @Override
    public String getFormattedValue(float value, AxisBase axis) {
        // avoid memory allocations here (for performance)
        return mFormat.format(value);
    }

    @Override
    public int getDecimalDigits() {
        return digits;
    }
}

File: MPChartLib/src/main/java/com/github/mikephil/charting/formatter/DefaultAxisValueFormatter.kt
package com.github.mikephil.charting.formatter

import com.github.mikephil.charting.components.AxisBase
import java.text.DecimalFormat

open class DefaultAxisValueFormatter(digits: Int) : IAxisValueFormatter {
    /**
     * decimal format for formatting
     */
    protected var decimalFormat: DecimalFormat
    /**
     * Returns the number of decimal digits this formatter uses or -1, if unspecified.
     */
    /**
     * the number of decimal digits this formatter uses
     */
    var decimalDigits = 0
        protected set

    /**
     * Constructor that specifies to how many digits the value should be
     * formatted.
     */
    init {
        decimalDigits = digits
        val b = StringBuffer()
        for (i in 0 until digits) {
            if (i == 0) b.append(".")
            b.append("0")
        }
        decimalFormat = DecimalFormat("###,###,###,##0$b")
    }

    override fun getFormattedValue(value: Float, axis: AxisBase?): String? {
        // avoid memory allocations here (for performance)
        return decimalFormat.format(value.toDouble())
    }
}
